infrastructure/tooling/src/smoketest/checks/role.js

```diff
@@ -0,0 +1,25 @@
+const taskcluster = require('taskcluster-client');
+
+exports.tasks = [];
+exports.tasks.push({
+  title: 'Create role and expand smoketest (--target roles)',
+  requires: [],
+  provides: [
+    'target-roles',
+  ],
+  run: async () => {
+    let auth = new taskcluster.Auth(taskcluster.fromEnvVars());
+    let randomId = taskcluster.slugid();
+    let roleId = `project:taskcluster:smoketest:${randomId}`;
+    let payload = {
+      description: 'smoketest test',
+      scopes: ['project:taskcluster:smoketest:*'],
+    };
+    let expandPayload = {
+      scopes: [`assume:${roleId}`],
+    };
+    await auth.createRole(roleId, payload);
+    await auth.expandScopes(expandPayload);
```

Comment:
We'll need to compare the result to what's expected :)

Reply:
Yes, totally :).

Reply:
Yes, totally :).